### DISCUSSION TITLE
Is it possible to invoke AWS Sagemaker Endpoint in pyscript? Is it possible to invoke AWS S3 in pyscript?

### DISCUSSION THREAD

**Discussion by cosmosanalytics** [pyscript/pyscript#2048] [Proposals]
I want to use Pyscript to leverage the storage from AWS S3 and the program execution capability (such as GPU) from AWS Sakemaker Endpoint. 

I understand I currently can't use boto3 in Pyscript. But I understand I can call API in Pyscript (https://randomds.com/2022/07/15/call-an-api-from-pyscript/#google_vignette) and I can use AWS API gateway to access AWS S3 and AWS Sakemaker endpoint. 

Can someone provide some examples for how to do the above?

**Comment by WebReflection** [comment]
while in theory any API that works remotely via HTTP should be possible, if you do this from the browser it's very likely you need to share private keys on the front end too, which is an absolute **must not** for security and (your) billing concerns.

**Comment by cosmosanalytics** [comment]
I am just curious why in pyscript.com there is CHATGPT-demo, which also
requires authentication. I thought pyscript encrypts authentication.

Also the web sites I am developing is internal behind firewall for a small
business team. I just don't want the headaches of IT setting up server.

**Reply by WebReflection** [reply]
you could ask for a prompt to store in your `localStorage` (instead of making it publicly available) your credentials then use `fetch` API to communicate with *AWS* ... then your credentials won't be stored anymore.

The Chat GPT demo was using an account credential but I think it's been fixed or currently it's not using a single person secret keys .. the rule of thumb is: if it's a secret key, it should **NOT** be on the client side, unless stored after asking it and only locally so it cannot be shared.

**Comment by cosmosanalytics** [comment]
Thank you so much!
